### PR TITLE
Installer: Add shutdown

### PR DESCRIPTION
**Pull request by mook-as**
This shuts down the ollama server (that we spawn) on demand.

This fixes part 1 of rancher-sandbox/rancher-desktop-rdx-open-webui#4.
Part 2 already happens (we're reusing the code from that here).
Part 3 needs to be done in Rancher Desktop.

### CODE REVIEW

**Comment by jandubois** [comment]
> This shuts down the ollama server (that we spawn) on demand.

I noticed that the "that we spawn" part is not true:

1. I run `ollama serve` from a shell window.
2. I open the extension and interact with the model
3. I quit Rancher Desktop
4. When the Rancher Desktop window closes, `ollama` also exits and I'm back at the shell prompt.

**Comment by jandubois** [comment]
Testing on macOS; Ollama has been installed by upstream installer, but is not running.

Extension will start /usr/local/bin/ollama serve process when it is activated.

When I quit Rancher Desktop the ollama process continues to run (I assume because the executable lives in `/usr/local/bin` outside the extension directory).

**Comment by mook-as** [comment]
Yeah, there's no difference as far as we can tell whether we started it or somebody else did. It's possible that you have started using it outside of Rancher Desktop halfway through, and we shouldn't kill it.

**Comment by jandubois** [comment]
> Yeah, there's no difference as far as we can tell whether we started it or somebody else did. It's possible that you have started using it outside of Rancher Desktop halfway through, and we shouldn't kill it.

We can write a PID file into the extension directory when we start it. That way if the path does not match the extension directory, we can still check the pid. If it matches it is still ours to kill.

File: installer/main.go [hunk 1]
```diff
@@ -39,16 +41,9 @@ func main() {
...
-		case string(ModeInstall):
-			mode = ModeInstall
-		case string(ModeUninstall):
-			mode = ModeUninstall
-		case string(ModeCheck):
-			mode = ModeCheck
-		case string(ModeStart):
-			mode = ModeStart
-		default:
+		if i := slices.Index(allModes, Mode(s)); i > -1 {
```
Comment (jandubois) [inline installer/main.go:44]: I was thinking that `allModes` should be a `map[string]bool`, so you could check it with `allModes[s]`, but then you would have to use `maps.Keys(allModes)` elsewhere, so it doesn't really matter.

**Review by jandubois — APPROVED** [review]
Thanks, LGTM

**Pull request merged**